pkg/runner/runner.go

```diff
@@ -115,6 +119,9 @@ func (r *Runner) call(callCtx engine.Context, monitor Monitor, env []string, inp
 				Type:        EventTypeCallFinish,
 				Content:     *result.Result,
 			})
+			if err := recordStateMessage(result.State); err != nil {
```

Comment:
nit: should a failure to write this debug info cause this method to error out?

Reply:
Good point. let's make sure this not fail and log a warning.